diff -r -U3 cpp_redis.bak/includes/cpp_redis/network/io_service.hpp cpp_redis/includes/cpp_redis/network/io_service.hpp
--- cpp_redis.bak/includes/cpp_redis/network/io_service.hpp	2017-05-30 00:18:05.612368807 +0300
+++ cpp_redis/includes/cpp_redis/network/io_service.hpp	2017-05-30 00:18:33.315368019 +0300
@@ -1,5 +1,6 @@
 #pragma once
 
+#include <functional>
 #include <thread>
 #include <atomic>
 #include <unordered_map>
